src/utils/form-fare-basis-code.js

```diff
@@ -0,0 +1,21 @@
+module.exports = (airPricingInfo, coupon) => {
+  const couponTicketDesignator = coupon['air:TicketDesignator']
+    && coupon['air:TicketDesignator'].Value;
+
+  if (!airPricingInfo && couponTicketDesignator) {
+    return `${coupon.FareBasis}/${couponTicketDesignator}`;
+  } if (!airPricingInfo) {
+    return coupon.FareBasis;
+  }
+
+  const [fareInfoData] = Object.values(airPricingInfo['air:FareInfo'])
```

Comment:
Is designator from `airPricingInfo` more prioritized?

Reply:
correct, according to Martin's response: `I would say that you have to get AirPricingInfo and there parse FareInfo for each segment to get AirSegment/FareBasis/TicketDesignator. And AirSegments can be paired  with flight coupons` and test results